Frontend/src/Components/Faults/FaultsView.js

```diff
@@ -1,13 +1,334 @@
+import LowBatteryImage from "./Images/Low Battery.png";
+import MCUCheckImage from "./Images/MCU Check.png";
+import MPPTCurrentImage from "./Images/MPPT Current.png";
+import PowerWarningImage from "./Images/Power Warning.png";
+import BMSInputVoltageImage from "./Images/BMS Input Voltage.png";
+import PhysicalConnectionImage from "./Images/Physical Connection Lost.png";
+import WirelessCommsLostImage from "./Images/Wireless Comms Lost.png";
+import HighTemperatureImage from "./Images/High Temperature.png";
 
 export default function Faults(props) {
+    const getEStopString = () => {
+        let eStopStrings = [];
+
+        if(props.data?.battery_eStop[0]) {
+            eStopStrings.push("Battery");
+        }
+        if(props.data?.driver_eStop[0]) {
+            eStopStrings.push("Driver");
+        }
+        if(props.data?.external_eStop[0]) {
+            eStopStrings.push("External");
+        }
+
+        if(eStopStrings.length > 0) {
+            if(eStopStrings.length > 2) {
+                return eStopStrings[0] + ", " + eStopStrings[1].toLowerCase() + ", and " + eStopStrings[2].toLowerCase() + " E-stops were pressed";
+            } else if(eStopStrings.length > 1) {
+                return eStopStrings[0] + " and " + eStopStrings[1].toLowerCase() + " E-stops were pressed";
+            } else {
+                return eStopStrings[0] + " E-stop was pressed";
+            }
+        }
+        return "";
+    }
+
+    const checkBMSFailsafes = () => {
+        return (props.data?.voltage_failsafe[0]) || (props.data?.current_failsafe[0]) ||
+               (props.data?.supply_power_failsafe[0]) || (props.data?.memory_failsafe[0]) ||
+               (props.data?.relay_failsafe[0]);
+    }
+
+    const getBMSFailsafeString = () => {
+        let failsafeStrings = [];
+
+        if(props.data?.voltage_failsafe[0]) {
+            failsafeStrings.push("Voltage");
+        }
+        if(props.data?.current_failsafe[0]) {
+            failsafeStrings.push("Current");
+        }
+        if(props.data?.supply_power_failsafe[0]) {
+            failsafeStrings.push("Supply");
+        }
+        if(props.data?.memory_failsafe[0]) {
+            failsafeStrings.push("Memory");
+        }
+        if(props.data?.relay_failsafe[0]) {
+            failsafeStrings.push("Relay");
+        }
+
+        if(failsafeStrings.length > 0) {
+            if(failsafeStrings.length > 2) {
+                return failsafeStrings[0] + ", " + failsafeStrings[1].toLowerCase() + ", and " + failsafeStrings[2].toLowerCase() + " failsafes";
+            } else if(failsafeStrings.length > 1) {
+                return failsafeStrings[0] + " and " + failsafeStrings[1].toLowerCase() + " failsafes";
+            } else {
+                return failsafeStrings[0] + " failsafe";
+            }
+        }
+        return "";
+    }
+
+    const checkTemps = () => {
+        return (props.data?.motor_temp[0] > 100) || (props.data?.driverIO_temp[0] > 100) ||
```

Comment:
Can we set the magic numbers as a constant stored somewhere? Don't wanna go find all the instances of sensor bounds whenever they needa change